src/cargo/sources/registry/index.rs

```diff
@@ -689,6 +704,19 @@ impl<'a> SummariesCache<'a> {
         if *first_byte != CURRENT_CACHE_VERSION {
             bail!("looks like a different Cargo's cache, bailing out");
         }
+        let index_v_bytes = rest
+            .get(..4)
+            .ok_or_else(|| anyhow::anyhow!("cache expected 4 bytes for index version"))?;
+        let index_v = u32::from_le_bytes(index_v_bytes.try_into().unwrap());
+        if index_v > INDEX_V_MAX {
```

Comment:
I think that this check may actually want to be `!=` instead of `>` perhaps? If Cargo version 3 reads a cache file from Cargo version 2, I think that's the case I was worried about? (where version 2 didn't cache any version 3 entries in this summary file)

Reply:
Yes... 😡 😡  sorry, been making several elementary mistakes.